app/src/main/java/com/keylesspalace/tusky/util/Either.kt

```diff
@@ -21,27 +21,29 @@ package com.keylesspalace.tusky.util
  * Class to represent sum type/tagged union/variant/ADT e.t.c.
  * It is either Left or Right.
  */
-sealed class Either<out L, out R> {
-    data class Left<out L, out R>(val value: L) : Either<L, R>()
-    data class Right<out L, out R>(val value: R) : Either<L, R>()
+sealed interface Either<out L, out R> {
+    data class Left<out L, out R>(val value: L) : Either<L, R>
+    data class Right<out L, out R>(val value: R) : Either<L, R>
 
-    fun isRight() = this is Right
+    fun isRight(): Boolean = this is Right
 
-    fun isLeft() = this is Left
+    fun isLeft(): Boolean = this is Left
 
-    fun asLeftOrNull() = (this as? Left<L, R>)?.value
+    fun asLeftOrNull(): L? = (this as? Left<L, R>)?.value
 
-    fun asRightOrNull() = (this as? Right<L, R>)?.value
+    fun asRightOrNull(): R? = (this as? Right<L, R>)?.value
 
     fun asLeft(): L = (this as Left<L, R>).value
 
     fun asRight(): R = (this as Right<L, R>).value
 
-    inline fun <N> map(crossinline mapper: (R) -> N): Either<L, N> {
-        return if (this.isLeft()) {
-            Left(this.asLeft())
-        } else {
-            Right(mapper(this.asRight()))
+    companion object {
+        inline fun <L, R, N> Either<L, R>.map(crossinline mapper: (R) -> N): Either<L, N> {
+            return if (this.isLeft()) {
+                Left(this.asLeft())
+            } else {
+                Right(mapper(this.asRight()))
+            }
```

Comment:
> 'inline' modifier is not allowed on virtual members. Only private or final members can be inlined

Reply:
One more question here, seems no need to add `crossinline` for `mapper`, is there any specific reason to keep it?

Reply:
Addressing the fix to #4348.